### PR TITLE
Updated Homework #2 and Design Warm-Up #1 for 2021

### PR DESCRIPTION
**Pull request by borjasotomayor**
Homework #2 and Design Warm-Up #1 have only undergone some minor updates, mostly to update some terminology (lab vs homework, master vs main, etc.). For Homework #2 I also made a minor change to the starter code (see https://github.com/uchicago-cmsc22000/hw2-starter-code/commit/90eae34db98e1bc86b644ab84e8f64f19030fa31), and verified the code compiles correctly in the department's new Ubuntu 20.04 environment.

### CODE REVIEW

File: website/hw/hw2.rst [hunk 1]
```diff
@@ -1,31 +1,35 @@
...
-repository for each assignment). Like the previous lab, we will provide
-you with an *invitation URL* that will allow you sign up for the lab
+In this homework, you will be using a different repository than the one you
+used in the previous homework (in general, GitHub will create a new
+repository for each assignment). Like the previous homework, we will provide
+you with an *invitation URL* that will allow you sign up for the homework
 assignment on GitHub, and which will result in the creation of a
-repository called ``2020-labN-GITHUB_USERNAME`` inside our
-``cmsc22000-labs`` organization on GitHub.
+repository called ``2021-hw2-GITHUB_USERNAME``` inside our
```
Comment (lilyehsani) [inline website/hw/hw2.rst:21]: There is an extra ` after USERNAME

File: website/hw/hw2.rst [hunk 2]
```diff
@@ -58,13 +62,13 @@ you can ignore it for now.
...
-not need to understand how the code is built to complete this lab.
+not need to understand how the code is built to complete this homework.
 
 The library currently provides two modules:
 
--  A Point module (``include/point.h' and``\ src/point.c`) that provides
+-  A Point module (``include/point.h`` and ``src/point.c``) that provides
    a number of operations on two-dimensional points
--  A Polygon module (``include/polygon.h' and``\ src/polygon.c`) that
+-  A Polygon module (``include/polygon.h' and ``src/polygon.c``) that
```
Comment (lilyehsani) [inline website/hw/hw2.rst:71]: This line is missing a `` after polygon.h'

File: website/project/warmup.rst [hunk 3]
```diff
@@ -1,4 +1,258 @@
...
+by a lot of other teams in the Boardmaster 3000 project. For example,
+when our advanced Chess AI, fondly nicknamed “The Chessinator 3000”,
+needs to know whether a piece in position (x0, y0) can move to position
+(x1, y1), it will be calling your code.
+
+Take into account that you are only going to be *designing* these
+modules. So, you will be focusing only on specifying the *interface* of
+those modules, and not their implementation (with one exception: you
+will have to provide a specification of any internal data structures you
+expect to use)
```
Comment (lilyehsani) [inline website/project/warmup.rst:99]: Missing a period end of this line

File: website/project/warmup.rst [hunk 4]
```diff
@@ -1,4 +1,258 @@
...
+
+-  Loading a game from disk
+-  Saving a game to disk
+-  Determining the following about the state of the game:
+
+   -  What piece (if any) is in position *x, y*?
+   -  What are the possible moves for the piece in position *x, y*?
+   -  Can piece in position (x0, y0) move to position (x1, y1)?
+   -  What piece (if any) will the piece in position (x0, y0) take if it
+      moves to position (x1, y1)
```
Comment (lilyehsani) [inline website/project/warmup.rst:111]: Missing a question mark end of this line

File: website/project/warmup.rst [hunk 5]
```diff
@@ -1,4 +1,258 @@
...
+       /* This function adds two numbers */
+       int add(int x, int y);
+
+   math.c:
+
+       /* See math.h*/
+       int add(int x, int y)
+       {
+           /* TODO */
+           Return 0;
```
Comment (lilyehsani) [inline website/project/warmup.rst:149]: I think the r in ``Return`` should be lowercase here

**Review by lilyehsani — CHANGES_REQUESTED** [review]
I made some minor comments about missing punctuation for clarity. Overall, the writeup looks great and compiles in the CSIL Linux environment (I ran `make`, `make samples`, `sample/polygon.h`, and `make tests`). I believe everything is written clearly and I do not anticipate any specific questions that we could address further.

File: website/project/warmup.rst [hunk 6]
```diff
@@ -1,4 +1,258 @@
...
+`chess <https://en.wikipedia.org/wiki/Chess>`__,
+`checkers <https://en.wikipedia.org/wiki/Draughts>`__, and some
+`variants of
+chess <https://en.wikipedia.org/wiki/List_of_chess_variants>`__)
+However, we are following an incremental software development model, so
+we are going to start by supporting only chess. This will give us a
+chance to get some initial feedback on the Boardmaster 3000 before we
+move on to supporting other games.
+
+What’s that you say? There’s a lot of chess games out there already? Ah,
```
Comment (firatciftci) [inline website/project/warmup.rst:23]: I think this should be changed to "There are a lot of chess games..."

**Review by firatciftci — CHANGES_REQUESTED** [review]
I've added a comment mentioning a small grammatical request, but apart from that everything seems to be good to go!

**Comment by borjasotomayor** [comment]
Thanks, I've pushed the fixes you suggested.

**Pull request merged**